Adding query for scala highlights

forestmvey opened the pull request:
## Description

Adding query  for scala highlights that adhere to [these rules](https://github.com/tonsky/sublime-scheme-alabaster#motivation)

## Screenshot

_***Note*** &mdash; I disable the comment highlights in my config_

<img width="978" height="655" alt="scala-alabaster-highlights" src="https://github.com/user-attachments/assets/da18442d-5957-4850-a1bf-0448ced2281a" />


## Example code

```scala
// single-line comment
/*
  multi-line comment
*/
package example.app
// Global object
object HighlightTest {
  val Pi: Double = 3.14159
  val Greeting: String = "Hello, Alabaster!"
  val IsScalaFun: Boolean = true
  val blah = 1 < 2
  var y = "hello"


  def add(x: Int, y: Int): Int = {
    x + y
  }
  def main(args: Array[String]): Unit = {
    println(Greeting)
  }
}
class User(val name: String, val age: Int) {
  def greet(): Unit = {
    println(s"Hello, my name is $name and I am $age years old.")
  }
}
trait Greetable {
  def greet(): Unit
}

```

p00f commented:
class member variables should not be highlighted: sublime text looks like this:
<img width="892" height="833" alt="image" src="https://github.com/user-attachments/assets/f0149d02-aaa6-4ee3-8ec2-186e0407835c" />

forestmvey commented:
> class member variables should not be highlighted: sublime text looks like this: <img alt="image" width="892" height="833" src="https://private-user-images.githubusercontent.com/36493671/486467544-f0149d02-aaa6-4ee3-8ec2-186e0407835c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTcyNjQ4NzAsIm5iZiI6MTc1NzI2NDU3MCwicGF0aCI6Ii8zNjQ5MzY3MS80ODY0Njc1NDQtZjAxNDlkMDItYWFhNi00ZWUzLThlYzItMTg2ZTA0MDc4MzVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTA3VDE3MDI1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI5Yzc0ZDU0MGM3ZGRlYTZhMWYxNDI0OGQxODRlYjI2NGY1MDY3OTA1ODhmOWI3ZmY1Y2VmNzQ1ZjE2OGVlNzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8dFsLOi0tHxNjX3uOh6mlvJCjuCNF3T0_0pqUhtFmb8">

Addressed in [commit-2](https://github.com/p00f/alabaster.nvim/pull/22/commits/7789a0ab89ec31856b46e6a4fc0c482697ad58ca) and I updated the screenshot

forestmvey commented:
> What is `var_definition`? I don't see it when I `:InspectTree` your example snippet.

Good catch, that was leftover from member variables highlighting. I removed it and updated the code example with a var definition.

```scala
  var y = "hello"
```

p00f commented:
also sorry i don't want to be a prude but your commit email looks sus (especially when i go to nixos.com in the browser)

p00f commented:
Merged, thanks!
